packages/template-retail-react-app/app/hooks/use-auth-modal.js

```diff
@@ -38,69 +46,104 @@ export const AuthModal = ({
     onPasswordResetSuccess = noop,
+    isOpen,
+    onOpen,
+    onClose,
     ...props
 }) => {
     const {formatMessage} = useIntl()
-    const customer = useCustomer()
+    const customerId = useCustomerId()
+    const {isRegistered} = useCustomerType()
+    const customer = useCustomer(
+        {customerId},
+        {enabled: !!customerId && isRegistered}
+    )
     const navigate = useNavigation()
     const [currentView, setCurrentView] = useState(initialView)
     const form = useForm()
     const submittedEmail = useRef()
     const toast = useToast()
+    const login = useShopperLoginHelper(ShopperLoginHelpers.LoginRegisteredUserB2C)
+    const register = useShopperLoginHelper(ShopperLoginHelpers.Register)
+    
+    // TODO: simplify the args to remove action
+    const getResetPasswordToken = useShopperCustomersMutation({action: ShopperCustomersMutations.GetResetPasswordToken})
 
     const submitForm = async (data) => {
         form.clearErrors()
 
-        return {
-            login: handleLogin,
-            register: handleRegister,
-            password: handleResetPassword
-        }[currentView](data)
-    }
-
-    const handleLogin = async (data) => {
-        try {
-            await customer.login(data)
-        } catch (error) {
-            const message = /invalid credentials/i.test(error.message)
-                ? formatMessage({
-                      defaultMessage:
-                          "Something's not right with your email or password. Try again.",
-                      id: 'auth_modal.error.incorrect_email_or_password'
-                  })
-                : formatMessage(API_ERROR_MESSAGE)
-            form.setError('global', {type: 'manual', message})
-        }
-    }
+        // mutation is cached by default?
+        // TODO: should we disable mutation cache by default?
+        login.reset()
+        register.reset()
 
-    const handleRegister = async (data) => {
-        try {
-            await customer.registerCustomer(data)
+        const onLoginSuccess = () => {
             navigate('/account')
-        } catch (error) {
-            form.setError('global', {type: 'manual', message: error.message})
         }
-    }
 
-    const handleResetPassword = async ({email}) => {
-        try {
-            await customer.getResetPasswordToken(email)
-            // Execute action to be perfromed on successful passoword reset
-            await onPasswordResetSuccess()
-            submittedEmail.current = email
-        } catch (error) {
-            form.setError('global', {type: 'manual', message: error.message})
-        }
+        return {
+            login: (data) =>
+                login.mutateAsync(
+                    {username: data.email, password: data.password},
+                    {
+                        onSuccess: onLoginSuccess,
+                        onError: (error) => {
+                            const message = /Unauthorized/i.test(error.message)
```

Comment:
As SCAPI/OCAPI errors every localized? Can we reliably expect this message regardless of the locale we're using?

Reply:
`error.message` is actually not the SCAPI error. The error is the [ResponseError](https://github.com/SalesforceCommerceCloud/commerce-sdk-isomorphic/blob/main/src/static/responseError.ts#L17) thrown by the isomorphic client. The message is always in the format of `${statusCode} ${statusText}` and `Unauthorized` is the status text for 401.